src/dmd/interop.d

```diff
@@ -0,0 +1,31 @@
+/+
+Example usage:
+extern (C++) Dstring funLib(Dstring input)
+{
+    import std.string : toUpper;
+    import std.conv : to;
+    return input.toNative.toUpper.to!Dstring;
+}
++/
+
+extern (C++) struct Darray(T)
```

Comment:
If you're going to create a wrapper type like this you'll probably need to do it in the same way that I've done in `core.sentinel`.  Namely, you'll need to support implicit conversion from:
```
Darray!(immutable(char)) > Darray!(const(char))
Darray!char > Darray!(const(char))
```

You may also need to support implicit conversion to/from native D arrays, but implicit conversion between multiple types requires multiple alias this.

Take a look at `core.sentinel` to see how this is done.